contrib/opencensus-ext-stackdriver/opencensus/ext/stackdriver/stats_exporter/__init__.py

```diff
@@ -131,209 +130,174 @@ def options(self):
-            start_time.nanos = int((timestamp_start - start_secs) * 1e9)
-
-            time_series_list.append(series)
-
-        return time_series_list
-
-    def create_metric_descriptor(self, view):
-        """ it creates a MetricDescriptor
-        for the given view data in Stackdriver Monitoring.
-        An error will be raised if there is
-        already a metric descriptor created with the same name
-        but it has a different aggregation or keys.
-        """
-        view_measure = view.measure
-        view_aggregation = view.aggregation
-        view_name = view.name
-
-        metric_type = namespaced_view_name(view_name,
-                                           self.options.metric_prefix)
-        value_type = None
-        unit = view_measure.unit
-        metric_desc = monitoring_v3.enums.MetricDescriptor
-        agg_type = aggregation.Type
-
-        # Default metric Kind
-        metric_kind = metric_desc.MetricKind.CUMULATIVE
-
-        if view_aggregation.aggregation_type is agg_type.COUNT:
-            value_type = metric_desc.ValueType.INT64
-            # If the aggregation type is count
-            # which counts the number of recorded measurements
-            # the unit must be "1", because this view
-            # does not apply to the recorded values.
-            unit = str(1)
-        elif view_aggregation.aggregation_type is agg_type.SUM:
-            if isinstance(view_measure, measure.MeasureInt):
-                value_type = metric_desc.ValueType.INT64
-            if isinstance(view_measure, measure.MeasureFloat):
-                value_type = metric_desc.ValueType.DOUBLE
-        elif view_aggregation.aggregation_type is agg_type.DISTRIBUTION:
-            value_type = metric_desc.ValueType.DISTRIBUTION
-        elif view_aggregation.aggregation_type is agg_type.LASTVALUE:
-            metric_kind = metric_desc.MetricKind.GAUGE
-            if isinstance(view_measure, measure.MeasureInt):
-                value_type = metric_desc.ValueType.INT64
-            if isinstance(view_measure, measure.MeasureFloat):
-                value_type = metric_desc.ValueType.DOUBLE
+    def create_time_series_list(self, metric):
+        tsl = []
+        for ts in metric.time_series:
+            tsl.append(self._convert_series(metric, ts))
+        return tsl
+
+    def _convert_series(self, metric, ts):
+        """Convert an OC timeseries to a SD series."""
+        series = monitoring_v3.types.TimeSeries()
+        series.metric.type = namespaced_view_name(
+            metric.descriptor.name, self.options.metric_prefix)
+
+        series.metric.labels[OPENCENSUS_TASK] = get_task_value()
+
+        for key, val in zip(metric.descriptor.label_keys, ts.label_values):
+            if val.value is not None:
+                safe_key = sanitize_label(key.key)
+                series.metric.labels[safe_key] = val.value
+
+        set_monitored_resource(series, self.options.resource)
```

Comment:
Outside the scope of this pr, but I'm curious what the motivation is to do the monitored resource detection every time we export every series. Instead would it make sense to detect and convert the labels once when the exporter is created? Then we can just set it on each series each time.

Reply:
nvm, #427 addresses this